### PR TITLE
Add migration guide for legacy script projects

### PR DESCRIPTION
**Pull request by willeastcott**
Add a migration guide for upgrading a legacy script project to use Scripts 2.0.

image

### CODE REVIEW

File: content/en/user-manual/scripting/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+And the following message in the Editor:
+
+![Editor Legacy Script][3]
+
+If you want to continue to work on legacy script projects, we recommend that you migrate them to the current format. Unfortunately, there is no automated migration process for this. Instead, it is a manual process, but this guide will walk you through it.
+
+### Step 1 - Create a New Project
+
+We will begin by creating a new, blank project to receive the update project. Delete the four entities created by default below the scene root entity.
```
Comment (yaustar) [inline content/en/user-manual/scripting/migration-guide.md:23]: "We will begin by creating a new, blank project to transfer assets and code into."

File: content/en/user-manual/scripting/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+### Step 3 - Transfer Scripts
+
+Download the scripts from your legacy script project via the Project Dashboard:
+
+![Download Scripts][5]
+
+If your legacy project is connected to a GitHub repo, download your scripts from there.
+
+You can now upload the legacy scripts to your new project.
```
Comment (yaustar) [inline content/en/user-manual/scripting/migration-guide.md:41]: Maybe show a screenshot of upload process in the assets panel.

File: content/en/user-manual/scripting/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+![Download Scripts][5]
+
+If your legacy project is connected to a GitHub repo, download your scripts from there.
+
+You can now upload the legacy scripts to your new project. 
+
+### Step 4 - Update Scripts to Current Format
+
+The newly transferred scripts in your new project will now need to be updated to the current format. Here is an example, showing the legacy format:
```
Comment (yaustar) [inline content/en/user-manual/scripting/migration-guide.md:45]: No need for 'newly'.

File: content/en/user-manual/scripting/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    this.on("destroy", function () {
+
+    });
+};
+```
+
+### Step 5 - Transfer Scene Hierarchy
+
+Next, we will transfer the legacy project's scene hierarchy across. The PlayCanvas Editor supports a copy and paste operation between two Editor instances. However, this operation fails if legacy script components are in the selection. Therefore, you should first delete all script components from your legacy script project. To do this, select all entities with legacy script components. You can do this by running the following JavaScript in the browser's JavaScript console:
+
```
Comment (yaustar) [inline content/en/user-manual/scripting/migration-guide.md:132]: Show a screenshot of the dev console?

File: content/en/user-manual/scripting/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+}  
+```
+
+You can then hit the delete button in the Inspector:
+
+![Delete Script Components][6]
+
+Now you can successfully copy and paste your game's hierarchy from the read-only project to the new destination project.
+
+With this done, you can hit CTRL+Z to undo the previous deletion of your legacy script components.
```
Comment (yaustar) [inline content/en/user-manual/scripting/migration-guide.md:150]: Mac users with use CMD+Z

**Review by yaustar — APPROVED** [review]

File: content/en/user-manual/scripting/migration-guide.md (hunk 1 above)
Comment (willeastcott) [reply content/en/user-manual/scripting/migration-guide.md:23]: Done.

File: content/en/user-manual/scripting/migration-guide.md (hunk 3 above)
Comment (willeastcott) [reply content/en/user-manual/scripting/migration-guide.md:45]: Done.

File: content/en/user-manual/scripting/migration-guide.md (hunk 4 above)
Comment (willeastcott) [reply content/en/user-manual/scripting/migration-guide.md:132]: Done.

File: content/en/user-manual/scripting/migration-guide.md (hunk 5 above)
Comment (willeastcott) [reply content/en/user-manual/scripting/migration-guide.md:150]: Done.

File: content/en/user-manual/scripting/migration-guide.md (hunk 2 above)
Comment (willeastcott) [reply content/en/user-manual/scripting/migration-guide.md:41]: Not sure how to do this unless it's an animated GIF. I'd hope devs would know how to upload assets though (if they have legacy projects).

**Pull request merged**